lib/src/input_validation/validate_function.dart

```diff
@@ -36,46 +39,57 @@ class ValidateFunction {
       }
     }
 
+    return inputString;
+  }
+
+  void incrementCounter(){
+    if(isMultiParam)
+      this.multiParamCounter = this.multiParamCounter + 1;
+    else
+      this.counter = this.counter + 1;
+  }
+
+  bool testFunction(String input) {
+    List<String> inputString = initialize(input);
     currentState= new StartState(this);
 
-    for(int i = 0; i < inputString.length; i++){
+    for(int i = 0; i < inputString.length; i++) {
       //handle special negatives for complex functions
       if(RegExp(r'^-[a-z]+$').hasMatch(inputString[i]) && inputString[i].length > 1) {
         inputString[i] = inputString[i].substring(1); // remove the negative
       }
 
       if(RegExp(r'^-?[0-9]+(.[0-9]+)?$|^[𝜋𝑒]$', unicode: true).hasMatch(inputString[i]) || inputString[i].length == 1) {  // numbers or operands
-        this.counter = currentState.getNextState(inputString[i], counter);
+        if(this.isMultiParam)
+          this.multiParamCounter = currentState.getNextState(inputString[i], multiParamCounter, isMultiParam);
+        else
+          this.counter = currentState.getNextState(inputString[i], counter, isMultiParam);
 
-        if(currentState is ErrorState){
+        if(currentState is ErrorState)
           return false;
-        }
       }
-      else if(inputString[i].length == 2){
-        if(inputString[i] == "-("){
-          // handle expression special case
-          // Increment the counter and update state
-          this.counter = this.counter + 1;
+      else if(inputString[i].length == 2) {
+        if(inputString[i] == "-(") {
+          // handle expression special case and Increment the counter and update state
+          incrementCounter();
           currentState = new OpenSubExpressionState(this);
         }
-        else if(lengthTwoFunc.contains(inputString[i]) == false){
+        else if(lengthTwoFunc.contains(inputString[i]) == false)
           return false;
-        }
       }
-      else if(inputString[i].length == 3){
-        if(lengthThreeFunc.contains(inputString[i]) == false){
+      else if(inputString[i].length == 3) {
+        if(!lengthThreeFunc.contains(inputString[i]) && !lengthThreeMultiParamFunc.contains(inputString[i]) )
           return false;
-        }
+        else if(lengthThreeMultiParamFunc.contains(inputString[i]))
+          isMultiParam = true;
       }
-      else if(inputString[i].length == 4){
-        if(lengthFourFunc.contains(inputString[i]) == false){
+      else if(inputString[i].length == 4) {
+        if(lengthFourFunc.contains(inputString[i]) == false)
           return false;
-        }
       }
-      else if(inputString[i].length == 5){
-        if(lengthFiveFunc.contains(inputString[i]) == false){
+      else if(inputString[i].length == 5) {
```

Comment:
Combine all these similar length 2, 3, 4, 5 checks into just one check that checks one list.

Reply:
I think it is useful to make sure we're not checking everything and increase computation time and complexity.

Reply:
Also, 3 is still different